### PR TITLE
Connect support

### PR DESCRIPTION
**Pull request by alexdavid**
@charlierudolph @kevgo 

This makes exprestive work with any connect-based app and updates the tests to prove it :smile:

### CODE REVIEW

**Comment by alexdavid** [comment]
We should probably update the readme to mention that express isn't strictly required

**Comment by charlierudolph** [comment]
Was just going to mention that. Code looks good

File: src/exprestive.coffee [hunk 1]
```diff
@@ -51,7 +51,8 @@ class Exprestive
 
   # Middleware to set reverse routes on req.locals
   setReverseRoutesOnReqLocals: (req, res, next) =>
-    res.locals.routes = @reverseRoutes
+    if res.locals?
+      res.locals.routes = @reverseRoutes
```
Comment (kevgo) [inline src/exprestive.coffee]: Put the if-clause after the expression? It's not really code flow, just to guard against an edge case, so doesn't have to be so front and center.

``` coffeescript
res.locals.routes = @reverseRoutes if res.locals?
```

File: test/feature/step_definitions/setup_definitions.coffee [hunk 2]
```diff
@@ -1,17 +1,26 @@
 { expect } = require 'chai'
+coffee = require 'coffee-script'
 
 
 module.exports = ->
 
   @Given /^an exprestive app using defaults$/, (done) ->
-    @createExprestiveApp '', (err) =>
+    @createExprestiveApp library: 'connect', (err) =>
+      return done err if err
```
Comment (kevgo) [inline test/feature/step_definitions/setup_definitions.coffee:9]: All these ugly and redundant error checks in tests! How do you feel about me adding some [Nitroglycerin](https://github.com/Originate/nitroglycerin) here?

File: test/feature/support/helpers.coffee [hunk 3]
```diff
@@ -14,18 +14,19 @@ class Helpers
 
 
   # Creates a new exprestive app in @appPath
-  createExprestiveApp: (optionsStr, done) ->
+  createExprestiveApp: ({library, exprestiveOptions}, done) ->
     serverContents = """
       # Initialize exprestive application
-      express = require 'express'
+      #{library} = require '#{library}'
```
Comment (kevgo) [inline test/feature/support/helpers.coffee:20]: What is this fancy new syntax?

Comment (kevgo) [inline test/feature/support/helpers.coffee:20]: And why do you put the `library` argument in a string here? Is it sometimes not a string?

File: test/feature/support/helpers.coffee [hunk 4]
```diff
@@ -14,18 +14,19 @@ class Helpers
...
       exprestive = require 'exprestive'
-      app = express()
-      app.use exprestive(#{optionsStr})
+      app = #{library}()
```
Comment (kevgo) [inline test/feature/support/helpers.coffee:22]: This line doesn't compile in the CoffeeScript shell on coffeescript.org.

File: test/feature/support/helpers.coffee (hunk 3 above)
Comment (alexdavid) [inline test/feature/support/helpers.coffee:20]: the whole `serverContents` is a string that generates a sample exprestive application. `library` is a string of either `express` or `connect` based on which http library we want to test exprestive with

File: test/feature/step_definitions/setup_definitions.coffee (hunk 2 above)
Comment (alexdavid) [inline test/feature/step_definitions/setup_definitions.coffee:9]: If you want to send me a PR adding nitroglycerin I'll ship it

Comment (alexdavid) [inline test/feature/step_definitions/setup_definitions.coffee:9]: Or we could just use promises :smiling_imp:

File: src/exprestive.coffee (hunk 1 above)
Comment (alexdavid) [inline src/exprestive.coffee]: :+1:

File: test/feature/step_definitions/setup_definitions.coffee (hunk 2 above)
Comment (kevgo) [inline test/feature/step_definitions/setup_definitions.coffee:9]: Okay, what do you want to do here?

Comment (alexdavid) [inline test/feature/step_definitions/setup_definitions.coffee:9]: I'm going to ship this PR for now, I'll accept a future PR implementing nitroglycerine if you can get @charlierudolph to agree to it. Otherwise let's refactor these tests to use promises in a future PR

File: src/exprestive.coffee (hunk 1 above)
Comment (alexdavid) [inline src/exprestive.coffee]: Done
